### PR TITLE
Move the hashing from the serving

### PR DESCRIPTION
**Pull request by vagababov**
/assign @mattmoor

### CODE REVIEW

File: hash/hash.go [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+	})
+	return hd
+}
+
+// ChooseSubset consistently chooses n items from `from`, using
+// `target` as a seed value.
+// ChooseSubset is an internal function and presumes sanitized inputs.
+// TODO(vagababov): once initial impl is ready, think about how to cache
+// the prepared data.
+func ChooseSubset(from []string, n int, target string) sets.String {
```
Comment (mattmoor) [inline hash/hash.go:127]: I didn't realize that our input order mattered, but it does.  Should we have this take a sets.String (has better symmetry too)

File: hash/hash.go [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+		hd.nameLookup[hs] = f
+	}
+	// Sort for consistent mapping later.
+	sort.Slice(hd.hashPool, func(i, j int) bool {
+		return hd.hashPool[i] < hd.hashPool[j]
+	})
+	return hd
+}
+
+// ChooseSubset consistently chooses n items from `from`, using
```
Comment (mattmoor) [inline hash/hash.go:122]: We should really be better about documenting stuff in PKG.  Could I bother you for a detailed `doc.go` package comment with usage?

Comment (vagababov) [reply hash/hash.go:122]: Done. Not sure about the usage in doc...

**Pull request merged**